From: Yaniv Rosner <yanivr@broadcom.com>
Date: Wed, 22 May 2013 21:21:52 +0000
Subject: [PATCH 081/242] bnx2x: Enable `set_phys_id' for all functions
Patch-mainline: v3.11-rc1

Git-commit: 84b6f7456e8b88507dd85f988a9d350eb5af0e46
References: bsc#908684 FATE#317539


In Multi-function mode, all functions should be able to utilize said function;
There's no reason why only the link owner should be able to do so.

Signed-off-by: Yaniv Rosner <yanivr@broadcom.com>
Signed-off-by: Yuval Mintz <yuvalmin@broadcom.com>
Signed-off-by: Eilon Greenstein <eilong@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
@@ -2879,11 +2879,6 @@ static int bnx2x_set_phys_id(struct net_
 		return -EAGAIN;
 	}
 
-	if (!bp->port.pmf) {
-		DP(BNX2X_MSG_ETHTOOL, "Interface is not pmf\n");
-		return -EOPNOTSUPP;
-	}
-
 	switch (state) {
 	case ETHTOOL_ID_ACTIVE:
 		return 1;	/* cycle on/off once per second */
